pkg/system/phase2_creating.go

```diff
@@ -459,16 +459,19 @@ func (r *Reconciler) SetDesiredCoreApp() error {
 	podSpec.ServiceAccountName = "noobaa"
 	coreImageChanged := false
 
+	podSpec.Volumes = r.DefaultCoreApp.Volumes
+
 	for i := range podSpec.Containers {
 		c := &podSpec.Containers[i]
+		c.VolumeMounts = r.DefaultCoreApp.Containers[i].VolumeMounts
 		switch c.Name {
 		case "core":
 			if c.Image != r.NooBaa.Status.ActualImage {
 				coreImageChanged = true
 				c.Image = r.NooBaa.Status.ActualImage
 			}
 			// adding the missing Env variable from default container
-			util.MergeEnvArrays(&c.Env, &r.DefaultCoreApp.Env)
+			util.MergeEnvArrays(&c.Env, &r.DefaultCoreApp.Containers[0].Env)
```

Comment:
I think we can just move this out of the switch to line 467:
```suggestion
			util.MergeEnvArrays(&c.Env, &r.DefaultCoreApp.Containers[i].Env)
			r.setDesiredCoreEnv(c)			
```
WDYT?

Reply:
and of course remove 514-515

Reply:
Yes, it should work. I will update and test it. Thanks